internal/storage/storage_handle.go

```diff
@@ -67,8 +103,21 @@ func NewStorageHandle(ctx context.Context, clientConfig storageutil.StorageClien
 		clientOpts = append(clientOpts, option.WithEndpoint(clientConfig.CustomEndpoint.String()))
 	}
 
+	return storage.NewClient(ctx, clientOpts...)
+}
+
+// NewStorageHandle returns the handle of http or grpc Go storage client based on the
+// provided StorageClientConfig.ClientProtocol.
+// Please check out the StorageClientConfig to know about the parameters used in
+// http and gRPC client.
+func NewStorageHandle(ctx context.Context, clientConfig storageutil.StorageClientConfig) (sh StorageHandle, err error) {
 	var sc *storage.Client
-	sc, err = storage.NewClient(ctx, clientOpts...)
+	if clientConfig.ClientProtocol == mountpkg.GRPC {
+		sc, err = createGRPCClientHandle(ctx, &clientConfig)
+	} else {
+		sc, err = createHTTPClientHandle(ctx, &clientConfig)
```

Comment:
This way would be able to write unit test for wrong protocol (other than http1, http2 and grpc) but not to differentiate the http vs grpc.

Reply:
As discussed offline, we keep the check at both places:
(a) createGRPCClientHandle & createHTTPClientHandle - since it's hard to check the created client type (either http or grpc) as part of NewStorageHandle, hence added check in the createGRPCClientHandle & createHTTPClientHandle. This way, we can ensure, this is being called with right client-protocol.

(b) Also, added check in the NewStorageHandle to ensure, this is not being called with invalid client-protocol (neither of http1, http2 and grpc).